spec/models/reports/export_raw_spec.rb

```diff
@@ -0,0 +1,105 @@
+require "rails_helper"
+
+RSpec.describe Reports::ExportRaw do
+
+  let(:user) { FactoryGirl.create(:user) }
+  let(:facility) { FactoryGirl.create(:setup_facility, name: "My Facility", abbreviation: "MF") }
+  let(:account) { FactoryGirl.create(:account, :with_account_owner, owner: user) }
+
+  subject(:report) { described_class.new(**report_args) }
+
+  let(:report_args) do
+    {
+      action_name: "general",
+      facility: facility,
+      order_status_ids: [order_detail.order_status_id],
+      date_end: 1.day.from_now,
+      date_start: 1.day.ago,
+      date_range_field: "ordered_at",
+    }
+  end
+
+  describe "for an item" do
+    let(:item) { FactoryGirl.create(:setup_item, facility: facility) }
+    let(:order_detail) do
+      place_product_order(user, facility, item, account).tap do |od|
+        od.complete!
+        od.manually_priced!
+        od.update_attributes!(
+          quantity: 3,
+          actual_cost: BigDecimal("19.99"),
+          actual_subsidy: BigDecimal("9.99"),
+          estimated_cost: BigDecimal("39.99"),
+          estimated_subsidy: BigDecimal("29.99"),
+        )
+      end
+    end
+
+    it "exports correct number of line items" do
+      expect(report.to_csv.split("\n").length).to eq(2)
+    end
+
+    it "populates the report", :aggregate_failures do
+      is_expected.to have_column(Facility.model_name.human).with_value("My Facility (MF)")
+      is_expected.to have_column("Order").with_value(order_detail.to_s)
+      is_expected.to have_column("Ordered By").with_value(user.username)
+      is_expected.to have_column("First Name").with_value(user.first_name)
+      is_expected.to have_column("Last Name").with_value(user.last_name)
+      is_expected.to have_column("Quantity").with_value("3")
+      is_expected.to have_column("Estimated Cost").with_value("$39.99")
+      is_expected.to have_column("Estimated Subsidy").with_value("$29.99")
+      is_expected.to have_column("Estimated Total").with_value("$10.00")
+      is_expected.to have_column("Actual Cost").with_value("$19.99")
+      is_expected.to have_column("Actual Subsidy").with_value("$9.99")
+      is_expected.to have_column("Actual Total").with_value("$10.00")
+      is_expected.to have_column("Charge For").with_value("Quantity")
+    end
+
```

Comment:
idle speculation, is this slow because it's running the CSV report for each matcher, and if so, might the matcher be more like `is_expected.to have_columns("Order" => order_detail.to_s, "Ordered By" => user.username, ...` -- might be harder to read and not worth it, just wondering.